internal/cli/api/api.go

```diff
@@ -433,11 +443,17 @@ func remindUserToPinVersion(cmd *cobra.Command) {
 	}
 }
 
-func ensureVersionIsSupported(apiCommand shared_api.Command, version *string) {
-	for _, commandVersion := range apiCommand.Versions {
-		if commandVersion.Version == *version {
-			return
+func ensureVersionIsSupported(apiCommand shared_api.Command, versionString *string) {
+	version, err := shared_api.ParseVersion(*versionString)
+
+	// If the version is valid, check if it's supported
+	if err == nil {
+		for _, commandVersion := range apiCommand.Versions {
+			if commandVersion.Version.Equal(version) {
+				return
+			}
 		}
+		return
```

Comment:
I think you want to return the error if the ParseVersion failed or at least log the error to stderr to help debugging.

```suggestion
	// If the version is valid, check if it's supported
	if err != nil {
	    return err
	   }
		for _, commandVersion := range apiCommand.Versions {
			if commandVersion.Version.Equal(version) {
				return
			}
		}
		return
```